GVRf/Sample/gvrbulletsample/src/org/gearvrf/bulletsample/BulletSampleViewManager.java

```diff
@@ -0,0 +1,219 @@
+        addSphere(scene, 1.0f, 1.5f, 100.0f, -10.0f, 20.0f);
+    }
+
+    @Override
+    public void onStep() {
+        mBullet.doSimulation(1.0f / 60.0f, 10);
+        for (RigidBody body : rigidBodiesSceneMap.keySet()) {
+            if (body.geometry.shape.getType() == ShapeType.SPHERE_SHAPE_PROXYTYPE) {
+                rigidBodiesSceneMap
+                        .get(body)
+                        .getTransform()
+                        .setPosition(
+                                body.motionState.resultSimulation.originPoint.x,
+                                body.motionState.resultSimulation.originPoint.y,
+                                body.motionState.resultSimulation.originPoint.z);
+            }
+
+            if (body.geometry.shape.getType() == ShapeType.BOX_SHAPE_PROXYTYPE) {
+                rigidBodiesSceneMap
+                        .get(body)
+                        .getTransform()
+                        .setPosition(
+                                body.motionState.resultSimulation.originPoint.x,
+                                body.motionState.resultSimulation.originPoint.y,
+                                body.motionState.resultSimulation.originPoint.z);
+            }
+        }
+    }
+
+    private GVRSceneObject quadWithTexture(float width, float height,
+            String texture) {
+        FutureWrapper<GVRMesh> futureMesh = new FutureWrapper<GVRMesh>(
+                mGVRContext.createQuad(width, height));
+        GVRSceneObject object = null;
+        try {
+            object = new GVRSceneObject(mGVRContext, futureMesh,
+                    mGVRContext.loadFutureTexture(new GVRAndroidResource(
+                            mGVRContext, texture)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return object;
+    }
+
+    private GVRSceneObject meshWithTexture(String mesh, String texture) {
+        GVRSceneObject object = null;
+        try {
+            object = new GVRSceneObject(mGVRContext, new GVRAndroidResource(
+                    mGVRContext, mesh), new GVRAndroidResource(mGVRContext,
+                    texture));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return object;
+    }
+
+    /*
+     * Function to add a cube of unit size with mass at the specified position
+     * in Bullet physics world and scene graph.
+     */
+    private void addCube(GVRScene scene, float x, float y, float z, float mass) {
+        BoxShape boxShape = new BoxShape(new Vector3(0.5f, 0.5f, 0.5f));
+        Geometry boxGeometry = mBullet.createGeometry(boxShape, mass,
+                new Vector3(0.0f, 0.0f, 0.0f));
+        MotionState boxState = new MotionState();
+        boxState.worldTransform = new Transform(new Point3(x, y, z));
+        RigidBody boxBody = mBullet
+                .createAndAddRigidBody(boxGeometry, boxState);
+
+        GVRSceneObject cubeObject = meshWithTexture("cube.obj", "cube.jpg");
```

Comment:
Removed the redundant code. Also, for adding support for Bullet Physics with GearVRf will be a bigger task and there will be some design decisions we need to consider. Like, do we need to allocate physics world all the time whenever create Scene.